docs/tooling-consolidation.md

```diff
@@ -0,0 +1,82 @@
+# AspNetCore Tooling Consolidation
+
+## Objectives
+
+We want to consolidate portions of dotnet/aspnetcore-tooling into dotnet/aspnetcore in our ongoing effort to reduce overall build time end to end for the .NET Core SDK. As such, we will not require all of aspnetcore-tooling to be merged into aspnetcore. For example, tooling for VSMac and VSCode will remain in aspnetcore-tooling. The main motivation is to achieve the overall goals of repo consolidation to reduce the number of repository required to build the SDK while keeping the build in aspnetcore as simple as possible without needing to support build/test for tooling scenarios such as testing on VSCode. There is also an added benefit of maintaining the current development and release workflow for aspnetcore-tooling which is more "agile" than aspnetcore (e.g. faster PR builds, release cycles that synchronize with VS releases).
+
+To achieve this we will be migrating the following (and associated tests) from aspnetcore-tooling to aspnetcore:
+```
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X
+Microsoft.AspNetCore.Mvc.Razor.Extensions
+Microsoft.AspNetCore.Razor.Language
+Microsoft.AspNetCore.Razor.Tools
+Microsoft.CodeAnalysis.Razor
+Microsoft.NET.Sdk.Razor
```

Comment:
aspnetcore-tooling builds are official builds are not reliable, mostly due to Razor SDK tests if I remember correctly. For the last month, the passing rate of official 'master' builds is only 79%. I'm concerned this will significantly degrade aspnetcore reliability without test improvements **up front**. Have we made any progress in this area?

Reply:
I think @NTaylorMullen may have a better understanding of the reliability issues here, but from looking at the internal builds of master over the past month, there's a 85% pass rate: https://dnceng.visualstudio.com/internal/_pipeline/analytics/stageawareoutcome?definitionId=262&contextType=build
Specifically, the breakdown is as follows:
- 5 VS Code test failures
- 2 Validation failures 
- 1 Checkout failure
- 6 Build/Test failures
  - 1 agent timeout
  - 1 VS editor test failure
  - 1 VS live share test failure
  - 1 artifact upload failure
  - 2 Microsoft.NET.Sdk.Razor.Test failures

So it seems like only 2.5% of the builds were actually affected by Microsoft.NET.Sdk.Razor.Test test reliability. This area needs improvement but I wouldn't consider 2.5% a blocking issue for consolidation. I think our quarantine policy would be sufficient when we encounter these issues.

Reply:
Ya we've made huge strides in making our SDK tests better. AspNetCore has much larger flake problems than the SDK tests 😄. That being said, bringing them into the aspnetcore repo does add a layer of complexity which may or may not add flake.

@pranavkm do you believe that extra layer of complexity add to their flakiness?

Reply:
While they do increase the overall build time (and we can definitely spend some time tuning this if necessary), the SDK tests should be fairly stable. There's a class of the SDK tests that are problematic - tests that restore and we should be open to rethink how these get verified, but the rest should be pretty stable. The blazor-wasm also has tests that do msbuild integration tests and we've not seen a lot of flakiness here.

Reply:
I'll consider this as a non-issue then.